UPGRADING.md

```diff
@@ -218,111 +88,64 @@ domain.Create(context.Background(), &domain.CreateParams{
-    if ok {
-        // claims contain session information
-    } else {
-        // there is no active session (non-authenticated user)
-    }
-}
+### The type for failed API responses is now `*clerk.APIErrorResponse`
+
+The `v2` version of the library introduces a new type for API responses that contain errors.
+The new type is [clerk.APIErrorResponse](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#APIErrorResponse) and it replaces `clerk.ErrorResponse`.
+
+The `clerk.APIErrorResponse` type contains additional fields and provides access to more debugging information.
+
+```diff
+- org, err := client.Organizations().Create(clerk.CreateOrganizationParams{
+-     Name: "Acme Inc",
+- })
+- if err != nil {
+-     if errResp, ok := err.(*clerk.ErrorResponse); ok {
+-         // Access the API errors
+-         errResp.Errors
+-     }
+- }
++ ctx := context.Background()
++ clerk.SetKey("sk_live_XXX")
++ org, err := organization.Create(ctx, &organization.CreateParams{
++     Name: clerk.String("Acme Inc"),
++ })
++ if err != nil {
++     if apiErr, ok := err.(*clerk.APIErrorResponse); ok {
++         // Access the API errors and additional information
++         apiErr.TraceID
++         apiErr.Error()
++         apiErr.Response.RawJSON
++     }
++ }
 ```
 
-Here's the same HTTP server written for `v2`. Please note that only header based authentication with a bearer token is
-supported by the `RequireHeaderAuthorization` middleware. Cookie based authentication is not supported.
+### HTTP middleware
 
-```go
-import (
-    "net/http"
-
-    "github.com/clerk/clerk-sdk-go/v2"
-    clerkhttp "github.com/clerk/clerk-sdk-go/v2/http"
-)
-
-func main() {
-    clerk.SetKey("sk_live_XXX")
-    mux := http.NewServeMux()
-    mux.Handle("/session", clerkhttp.RequireHeaderAuthorization()(http.HandlerFunc(handleSession)))
-    http.ListenAndServe(":3000", mux)
-}
+The `clerk.WithSessionV2` and `clerk.RequireSessionV2` middleware functions from `v1` are replaced by [http.WithHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#WithHeaderAuthorization) and [http.RequireHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#RequireHeaderAuthorization) in `v2`
 
-func handleSession(w http.ResponseWriter, r *http.Request) {
-    sessionClaims, ok := clerk.SessionClaimsFromContext(r.Context())
-    if ok {
-        // claims contain session information
-    } else {
-        // there is no active session (non-authenticated user)
-    }
-}
-```
+Please note that as the name implies `WithHeaderAuthorization` and `RequireHeaderAuthorization` support only authentication with a bearer token.
+The token needs to be provided in the "Authorization" request header.
 
-If you're dealing with multiple Clerk API keys, you can pass a `jwks.Client` to the `RequireHeaderAuthorization` middleware.
+**Cookie based authentication is not supported at all by the `v2` version of the library.**
```

Comment:
💡 For tips and warnings you might benefit from the GHM blockquote syntax: https://docs.github.com/en/enterprise-cloud@latest/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

Reply:
Sweet, TIL and used that!